nixos/modules/tasks/network-interfaces-systemd.nix

```diff
@@ -59,8 +59,12 @@ let
       # more likely to result in interfaces being configured to
       # use DHCP when they shouldn't.
 
-      # When wait-online.anyInterface is enabled, RequiredForOnline really
-      # means "sufficient for online", so we can enable it.
+      # When wait-online.anyInterface is enabled (which it is by default due to
+      # systemd no longer considering "all interfaces not required for online"
+      # to be sufficient for being online
+      # <https://github.com/systemd/systemd/issues/29388>), RequiredForOnline
+      # really means "sufficient for online", so we can enable it.
```

Comment:
I thought the idea was that we should *remove* `RequiredForOnline=no` and instead just leave it unconfigured and let `--any` handle this?

Reply:
Leaving it as unconfigured is equivalent to setting it to true with the exception of if it is set to down by default. https://github.com/systemd/systemd/blob/117843fe95f091df1ddc87aaa60f1a2a8698d120/src/network/networkd-network.c#L282-L292

My thought process with leaving this code here is that if you *are* manually setting up networks *and* deliberately unset --any, you definitely don't want the default networks to be required to be considered online (e.g. in the case of a laptop with an Ethernet card that is not plugged in). But then again in that case I feel like you just want to turn off the default networks.

Feels like the global setting of --any is a clumsy tool from upstream, and I don't really know how to effectively use it.